nilearn/glm/tests/test_second_level.py

```diff
@@ -1178,3 +1178,26 @@ def test_second_level_contrast_computation_with_memory_caching():
         # Delete objects attached to files to avoid WindowsError when deleting
         # temporary directory (in Windows)
         del func_img, model, X, Y
+
+
+def test_second_lvl_dataframe_computation():
+    """Check that contrast can be computed when using dataframes as input.
+
+    See bug https://github.com/nilearn/nilearn/issues/3871
+    """
+    # 'abusing' write_fake_fmri_data_and_design
+    # to just generate a 3D image
+    shapes = ((7, 8, 9, 1),)
+    _, FUNCFILE, _ = write_fake_fmri_data_and_design(shapes)
```

Comment:
why not just use `write_fake_bold_img`?

Reply:
Because Rémi does not know his own utility functions...

More seriously: I still find it a bit hard to find the right data generation function when writing tests, so having some developer facing doc for this may help.

If we want to make it easier for contributors to write tests, this may prove useful.

This would probably help avoid our test suite getting WETter and WETter and we end up finding code duplicates in different test modules (see #3882).